crates/dekaf/src/connector.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::{bail, Context};
+use proto_flow::materialize;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+/// Configures the behavior of a whole dekaf task
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct DekafConfig {
+    /// Whether or not to expose topic names in a strictly Kafka-compliant format
+    /// for systems that require it. Off by default.
+    pub strict_topic_names: bool,
+    /// The password that will authenticate Kafka consumers to this task.
+    // TODO(jshearer): Uncomment when schemars 1.0 is out and we upgrade
+    // #[schemars(extend("secret" = true))]
+    pub token: String,
+}
+
+/// Configures a particular binding in a Dekaf-type materialization
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct DekafResourceConfig {
+    /// The exposed name of the topic that maps to this binding. This
+    /// will be exposed through the Kafka metadata/discovery APIs.
+    pub topic_name: String,
```

Comment:
This comment is _not_ to say that having dekaf's unary_materialize() understand Spec isn't a good idea too, if you can get that plumbed through reasonably. It's just not a requirement.

Reply:
> We'll likely want the x-table-name (?) annotation here so that the UI knows how to fill this out.
Huh, I didn't know that's how they got filled out. Does flowctl also do this clientside? I would have thought it'd be part of discovery, and was expecting to have to implement this in the Dekaf `unary_materialize` implementation. An annotation is certainly easier here though!

Reply:
It's filled out by the `agent` and by the `UI`. I wouldn't assume that `flowctl raw discover` has correct handling for it, but don't know for sure.

Reply:
Done -- I found `x-collection-name`, pretty sure that's the one